src/orion/core/utils/tree.py

```diff
@@ -327,7 +342,13 @@ def has_depth(node, children):
 
         return sum([node.item for node in nodes], [])
 
-    def map(self, function, node):
+    # TODO: Trying to type-annotate this function with overloads is a complete nightmare.
```

Comment:
Abort this file for this PR maybe?

Reply:
No! :stuck_out_tongue: 
Fixed in a5dac7573a3b7ff2bc879502cef1d28d849d2b66

Reply:
More seriously, this is used in PBT. I added the missing type annotations to PBT in 91a7c2221fbadde3397aa843e7901a08358c1dea, and pbt.py benefits from having fully-typed `TreeNode` (and `LineageNode` by extension)